test/e2e_node/container_lifecycle_pod_construction.go

```diff
@@ -124,28 +124,34 @@ func (o containerOutputList) String() string {
 
 // RunTogether returns an error the lhs and rhs run together
 func (o containerOutputList) RunTogether(lhs, rhs string) error {
-	lhsStart := o.findIndex(lhs, "Started", 0)
-	rhsStart := o.findIndex(rhs, "Started", lhsStart+1)
-
-	lhsFinish := o.findIndex(lhs, "Exiting", lhsStart+1)
-	rhsFinish := o.findIndex(rhs, "Exiting", rhsStart+1)
+	var startIdx int
+	for {
+		lhsStart := o.findIndex(lhs, "Started", startIdx)
+		// prepare for the next iteration
+		startIdx = lhsStart + 1
+		rhsStart := o.findIndex(rhs, "Started", lhsStart+1)
+
+		lhsFinish := o.findIndex(lhs, "Exiting", lhsStart+1)
+		rhsFinish := o.findIndex(rhs, "Exiting", lhsFinish+1)
```

Comment:
The same here - the code assumes that rhs always finished after lhs.

Reply:
yes, good point - I'll fix that